src/dmd/dinterpret.d

```diff
@@ -6464,98 +6525,59 @@ private Expression scrubReturnValue(const ref Loc loc, Expression e)
     else if (auto ale = e.isArrayLiteralExp())
     {
         ale.ownedByCtfe = OwnedBy.code;
-        if (Expression ex = scrubArray(loc, ale.elements))
+        if (auto ex = scrubArray(ale.elements))
             return ex;
     }
     else if (auto aae = e.isAssocArrayLiteralExp())
     {
         aae.ownedByCtfe = OwnedBy.code;
-        if (Expression ex = scrubArray(loc, aae.keys))
+        if (auto ex = scrubArray(aae.keys))
             return ex;
-        if (Expression ex = scrubArray(loc, aae.values))
+        if (auto ex = scrubArray(aae.values))
             return ex;
         aae.type = toBuiltinAAType(aae.type);
     }
     return e;
 }
 
-// Return true if every element is either void,
-// or is an array literal or struct literal of void elements.
-private bool isEntirelyVoid(Expressions* elems)
+/**************************************
+ * Transitively set all .ownedByCtfe to OwnedBy.cache
+ */
+private Expression scrubCacheValue(Expression e)
 {
-    foreach (e; *elems)
-    {
-        // It can be NULL for performance reasons,
-        // see StructLiteralExp::interpret().
-        if (!e)
-            continue;
-
-        if (!(e.op == TOK.void_) &&
-            !(e.op == TOK.arrayLiteral && isEntirelyVoid((cast(ArrayLiteralExp)e).elements)) &&
-            !(e.op == TOK.structLiteral && isEntirelyVoid((cast(StructLiteralExp)e).elements)))
-        {
-            return false;
-        }
-    }
-    return true;
-}
+    if (!e)
+        return e;
 
-// Scrub all members of an array. Return false if error
-private Expression scrubArray(const ref Loc loc, Expressions* elems, bool structlit = false)
-{
-    foreach (i, e; *elems)
+    Expression scrubArrayCache(Expressions* elems)
     {
-        // It can be NULL for performance reasons,
-        // see StructLiteralExp::interpret().
-        if (!e)
-            continue;
-
-        // A struct .init may contain void members.
-        // Static array members are a weird special case (bug 10994).
-        if (structlit &&
-            ((e.op == TOK.void_) ||
-             (e.op == TOK.arrayLiteral && e.type.ty == Tsarray && isEntirelyVoid((cast(ArrayLiteralExp)e).elements)) ||
```

Comment:
The `e.type.ty == Tsarray` is omitted in the `isVoid` function making the condition in scrubArray invalid.

Reply:
Want to fix it?

Reply:
https://github.com/dlang/dmd/pull/9547